### DISCUSSION TITLE
Sveltekit 1 + Cloudflare Pages

### DISCUSSION THREAD

**Discussion by ghost** [sveltejs/kit#8583] [General]
I just created a new Sveltekit 1 project (latest) and deployed to Cloudflare Pages.

Previously (pre Sveltekit 1) Ive used NODE_VERSION 7, however now I get errors for both NODE_VERSION 7 and NODE_VERSION 8, so had to use NODE_VERSION 6 to make it work.

Is this known? Sorry if Im missing something obvious. Obviously would be great to use Node 8.

ERROR LOG for NODE_VERSION 7:

16:49:25.748 | npm ERR! code EBADENGINE
-- | --
16:49:25.749 | npm ERR! engine Unsupported engine
16:49:25.751 | npm ERR! engine Not compatible with your version of node/npm: @sveltejs/kit@1.1.3
16:49:25.752 | npm ERR! notsup Not compatible with your version of node/npm: @sveltejs/kit@1.1.3
16:49:25.752 | npm ERR! notsup Required: {"node":"^16.14 \|\| >=18"}
16:49:25.752 | npm ERR! notsup Actual:   {"npm":"8.11.0","node":"v17.9.1"}
16:49:25.752 |  
16:49:25.752 | npm ERR! A complete log of this run can be found in:
16:49:25.752 | npm ERR!     /opt/buildhome/.npm/_logs/2023-01-18T15_49_25_528Z-debug-0.log
16:49:25.758 | Error during NPM install
16:49:25.761 | Failed: build command exited with code: 1
16:49:26.642 | Failed: an internal error occurred

ERROR LOG for NODE_VERSION 8:

16:51:34.985 | node: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by node)
-- | --
16:51:34.985 | node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by node)
16:51:34.986 | node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by node)
16:51:35.040 | nvm is not compatible with the npm config "prefix" option: currently set to ""
16:51:35.042 | Run `npm config delete prefix` or `nvm use --delete-prefix v18.13.0` to unset it.
16:51:35.043 | Failed to install node version '18'
16:51:35.047 | Failed: build command exited with code: 1
16:51:35.929 | Failed: an internal error occurred

**Comment by teemingc** [comment]
Try setting the env variable `NODE_VERSION` to `16`
The minimum node version for SvelteKit is 16 at the moment.

I followed these steps and am able to get a website running on Cloudflare Pages.

https://github.com/sveltejs/kit/tree/master/packages/adapter-cloudflare#deployment
https://developers.cloudflare.com/pages/framework-guides/deploy-a-svelte-site#sveltekit-cloudflare

**Reply by ghost** [reply]
Thanks. But were you able to build one with Node Version higher than 16?

**Reply by teemingc** [reply]
I don’t think Cloudflare supports 18 at this time
https://community.cloudflare.com/t/pages-build-failed-when-using-node-18-12-1-due-to-missing-glibc/433516/2

**Comment by SamuelEarl** [comment]
At the time of this writing this worked for me by adding a `NODE_VERSION` env variable and setting it to `20`.

**Reply by ABSanthosh** [reply]
2 SvelteKit apps did not have any problems but the third one has this issue. Thanks for the help. cfbr
